rocks-bin/src/list.rs

```diff
@@ -0,0 +1,45 @@
+use clap::Args;
+use eyre::{OptionExt as _, Result};
+use itertools::Itertools as _;
+use rocks_lib::{config::Config, tree::Tree};
+use text_trees::{FormatCharacters, StringTreeNode, TreeFormatting};
+
+#[derive(Args)]
+pub struct List {
+    #[arg(long)]
+    outdated: bool,
+    #[arg(long)]
+    porcelain: bool,
+}
+
+pub fn list(list_data: List, config: &Config) -> Result<()> {
+    let tree = Tree::new(
+        &config.tree,
+        config
+            .lua_version
+            .as_ref()
+            .ok_or_eyre("lua version not supplied!")?,
+    )?;
+    let available_rocks = tree.list();
+
+    // TODO(vhyrro): Add `outdated` support.
+
+    if list_data.porcelain {
+        for (name, versions) in available_rocks {
+            println!("{}: {}", name, versions.join(" "));
```

Comment:
suggestion (non-blocking): What if we ditch the luarocks "porcelain" output in favour of json?

Reply:
Love the sound of that idea. I'll keep the `--porcelain` name but change the internals to produce JSON instead.